packages/manager/apps/hub/src/data/api/apiOrder/apiOrder.ts

```diff
@@ -0,0 +1,85 @@
+import { aapi, v6 } from '@ovh-ux/manager-core-api';
+import {
+  OrderFollowUpResponse,
+  OrderResponseData,
+  OrderDetailsResponse,
+  OrderStatus,
+  OrderHistory,
+  LastOrderTrackingResponse,
+} from '@/types/order.type';
+import { WAITING_PAYMENT_LABEL } from './apiOrder.constants';
+
+export const getLastOrder: () => Promise<OrderResponseData> = async () =>
+  aapi.get('/hub/lastOrder').then(({ data }) => {
+    return data.data.lastOrder;
+  });
+
+export const getOrderDetails = async (
+  orderId: number,
+): Promise<OrderDetailsResponse> => {
+  const response = await v6.get(`/me/order/${orderId}/details`);
+  return response.data;
+};
+
+export const getOrderStatus = async (orderId: number): Promise<OrderStatus> => {
+  const response = await v6.get(`/me/order/${orderId}/status`);
+  return response.data as OrderStatus;
+};
+
+export const getOrderFollowUp = async (
+  orderId: number,
+): Promise<OrderFollowUpResponse> => {
+  const response = await v6.get(`/me/order/${orderId}/followUp`);
+  return response.data as OrderFollowUpResponse;
+};
+
+export const getCompleteHistory = async (
+  orderId: number,
+  orderStatus: string,
+  orderDate: string,
+): Promise<OrderHistory[]> => {
+  try {
+    const followUp = await getOrderFollowUp(orderId);
+    const history = followUp.flatMap((follow) => follow.history).reverse();
+
+    if (orderStatus === 'notPaid' && history.length === 0) {
```

Comment:
Why not use the constant here ?
This may require you to redo the definition of ERROR_STATUS